state_transfer/transfer/rsync/validation.go

```diff
@@ -0,0 +1,51 @@
+package rsync
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/konveyor/crane-lib/state_transfer/transfer"
+	errorsutil "k8s.io/apimachinery/pkg/util/errors"
+)
+
+const (
+	validatedPVCNameMaxLength = 63
+	pvcNameBadCharacters      = `[\\.]+`
+)
+
+// validatePVCList validates list of PVCs provided to rsync transfer
+func validatePVCList(pvcList transfer.PersistentVolumeClaimList) error {
+	srcNamespaces := pvcList.GetSourceNamespaces()
+	destNamespaces := pvcList.GetDestinationNamespaces()
+	if len(srcNamespaces) > 1 || len(destNamespaces) > 1 {
+		return fmt.Errorf("rsync transfer does not support migrating PVCs belonging to multiple source/destination namespaces")
```

Comment:
+1, I like this validation is in rsync package.

What we are saying here is that the interfaces themselves dont put the limitation of a single namespace, but rsync implementation does, and we can change that in the future as needed.